packages/framework/src/workflow.ts

```diff
@@ -27,9 +31,51 @@ export function workflow<
   workflowId: string,
   execute: Execute<T_Payload, T_Control>,
   workflowOptions?: WorkflowOptions<T_PayloadSchema, T_ControlSchema>
-): DiscoverWorkflowOutput {
+): Workflow<T_Payload> {
   const options = workflowOptions ? workflowOptions : {};
 
+  const apiClient = initApiClient(process.env.NOVU_SECRET_KEY as string);
+
+  const trigger: Workflow<T_Payload>['trigger'] = async (event) => {
+    if (!process.env.NOVU_SECRET_KEY) {
+      throw new MissingSecretKeyError();
+    }
+
+    let validatedData: T_Payload;
+    if (options.payloadSchema) {
+      const validationResult = await validateData(options.payloadSchema, event.payload);
```

Comment:
We've discussed this separately regarding DX, and I think it's a great opportunity to use the `const {data, error }` signature for this function instead of throwing an error. 

I'd like to hear your thoughts.

Reply:
If I'm consuming this `workflow.trigger()` method in my Typescript application, I would already have validated that my integration is correct via strong `payload` typings. Therefore if we're entering a state of an invalid `payload` being present, to me that's more of an exceptional state that warrants an error being thrown.

This is a wider conversation about client error handling that we need to look at. Right now, our `@novu/api` Speakeasy SDK is throwing an error here, I think we should maintain consistency with the opionated Speakeasy implementation.

Reply:
You are right about consistency. I added the Speakeasy team in the loop. We will probably release as such.